lib/searchyll/indexer.rb

```diff
@@ -111,20 +116,40 @@ def http_start
 
     # Prepare our indexing run by creating a new index.
     def prepare_index
+      if !@@mapping_file_path.nil? && File.exist?(@@mapping_file_path)
+        mapping_fields = JSON.parse(File.read(@@mapping_file_path))
+      else
+        mapping_fields = false
+      end
+      if !@@analysis_file_path.nil? && File.exist?(@@analysis_file_path)
+        analysis_fields = JSON.parse(File.read(@@analysis_file_path))
+      else
+        analysis_fields = false
+      end
       create_index = http_put("/#{elasticsearch_index_name}")
       create_index.body = {
-        index: {
-          number_of_shards:   configuration.elasticsearch_number_of_shards,
+        settings: {
+          number_of_shards: configuration.elasticsearch_number_of_shards,
           number_of_replicas: 0,
-          refresh_interval:   -1
+          refresh_interval: -1
         }
-      }.to_json # TODO: index settings
+      }.to_json
+      
+      # Add mapping fields to the index
+      if mapping_fields
+        create_index.class.instance_eval('attr_accessor :mappings')
+        create_index.mappings = mapping_fields
```

Comment:
What's the thought behind this method of mutating the class - rather than just changing how the hash is made that is passed to `create_index.body` - This seems odd to me, but maybe I could learn something about ruby. :)

Reply:
@drusellers Same thing as what said in this [comment](https://github.com/omc/searchyll/pull/34#discussion_r251320856): for unknown reasons in that moment i couldn't get it to works differently. I'm sure that yours ruby's knowledge are better then mine so i counting on it to solves this odd solution. Thank you!